### PR TITLE
Record additional failures in steps

### PR DESCRIPTION
**Pull request by ilyapuchka**
It's inconvenient when a test step fails to see failures only where the failure actually happened and not in which step of test it happened. With this change currently executed step, location is stored in state and when failure happens additional failure is recorded at the same location. This way it will be visible what step failed and what exact assert down the way caused it.

Failure in feature file:
screen shot 2018-08-02 at 20 40 15

Failure in steps definer:
screen shot 2018-08-02 at 20 42 35

### CODE REVIEW

File: Pod/Core/XCTestCase+Gherkin.swift [hunk 1]
```diff
@@ -40,6 +46,23 @@ class GherkinState {
...
+        XCTestObservationCenter.shared.addTestObserver(self)
+    }
+
+    deinit {
+        XCTestObservationCenter.shared.removeTestObserver(self)
+    }
+
+    func testCase(_ testCase: XCTestCase, didFailWithDescription description: String, inFile filePath: String?, atLine lineNumber: Int) {
+        guard let test = self.test else { return }
+        let file = "\(test.state.file!)"
```
Comment (deanWombourne) [inline Pod/Core/XCTestCase+Gherkin.swift]: While testing this to see how it affected the test results in Xcode I changed SanitySteps.swift:24 to be `XCTAssertTrue(false)` to force a fail, and it crashed on this line. For some reason the file is `nil`?

File: Pod/Core/StepDefiner.swift [hunk 2]
```diff
@@ -15,6 +15,16 @@ populate the step definitions before test runs
 open class StepDefiner: NSObject, XCTestObservation {
     public private(set) var test: XCTestCase
 
+    // file from where the step was invoked
+    public var file: StaticString {
```
Comment (deanWombourne) [inline Pod/Core/StepDefiner.swift]: Do these properties need to be public?

File: Pod/Core/XCTestCase+Gherkin.swift [hunk 3]
```diff
@@ -26,6 +26,12 @@ class GherkinState {
     // Used to track step nesting i.e. steps calling out to other steps
     var currentStepDepth: Int = 0
     
+    // file from where currently executed step was invoked
+    var file: StaticString!
```
Comment (deanWombourne) [inline Pod/Core/XCTestCase+Gherkin.swift]: Instead of two properties can we make this a single property i.e.

```swift
/// File and line where the currently executing step was invoked
var currentStepLocation: (file: StaticString, line: Uint)?
```

Doesn't make much difference to the implementation, but it does convey the intent of the property more clearly. It also makes sure that we don't accidentally set one without the other being set too :)

File: Pod/Native/NativeTestCase.swift [hunk 4]
```diff
@@ -121,8 +121,8 @@ extension XCTestCase {
         }
         
         if let background = feature.background {
-            background.stepDescriptions.forEach(self.performStep)
+            background.stepDescriptions.forEach({ self.performStep($0) })
```
Comment (deanWombourne) [inline Pod/Native/NativeTestCase.swift]: Why these changes? Nothing wrong with them, just interested why the style change :)

**Comment by ilyapuchka** [comment]
@deanWombourne all addressed

File: Pod/Native/NativeTestCase.swift (hunk 4 above)
Comment (deanWombourne) [reply Pod/Native/NativeTestCase.swift]: Ah, of course!

**Comment by ilyapuchka** [comment]
@deanWombourne I don't have permissions to merge

**Comment by deanWombourne** [comment]
@ilyapuchka Oh yea, sorry! I'll merge and release it now :)

**Comment by deanWombourne** [comment]
Hey. After a small fight with my mac I've released this as version 0.14.2
